### PR TITLE
example: dlang-app added

### PR DESCRIPTION
**Pull request by kassane**
## Summary

Add D (programming language) support, LDC2 (LLVM D Compiler) only!
url: https://ldc-developers.github.io/

## Impact

Another programming language alternative. Designed for 32-bit or higher devices.
It allows FFI C++ [`extern(C++)`] and C [`extern(C)`] code, in addition to turning the DRT(runtime)/GC (optional) using betterC or creating a miniDRT, if required.

## Testing

- https://github.com/kassane/nuttx-dlang-app

## Reference
- https://github.com/apache/nuttx/pull/12457

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -41,4 +41,6 @@ Make.dep
 .DS_Store
 build
 .ccls-cache
-compile_commands.json
\ No newline at end of file
+compile_commands.json
+# keep dlang source files
+!examples/hello_d/*.d
```
Comment (xiaoxiang781216) [inline .gitignore:46]: should we remove line 4?

Comment (lupyuen) [reply .gitignore:46]: @kassane `.ignore` has this at line 4. Perhaps we should remove this, so we don't need `!examples/hello_d/*.d`? 

https://github.com/apache/nuttx-apps/blob/1f443afcc95a87114c226747e7aa90df17f8d05c/.gitignore#L4

(I wonder why we are skipping `.d` files in the first place?)

**Comment by lupyuen** [comment]
Tested OK on macOS and `rv-virt:nsh` thanks!

Test Log: https://gist.github.com/lupyuen/e6fe3423e282ff8603de5f248bcb9ae5

File: .gitignore (hunk 1 above)
Comment (kassane) [reply .gitignore]: done!

**Review by lupyuen — APPROVED** [review]
Looks great, thank you so much! :-)

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by kassane** [comment]
> Tested OK on macOS and `rv-virt:nsh` thanks!

Nice! have you also tested arm (qemu or device)?

**Comment by xiaoxiang781216** [comment]
@kassane could you add/reuse a defconfig to enable hellod ci?

**Comment by lupyuen** [comment]
> ```shell
> arm-none-eabi-ld: unwind-arm.c:(.text.get_eit_entry+0x6c): undefined reference to `__exidx_end'
> arm-none-eabi-ld: unwind-arm.c:(.text.get_eit_entry+0x70): undefined reference to `__exidx_start'
> arm-none-eabi-ld: unwind-arm.c:(.text.get_eit_entry+0x74): undefined reference to `__exidx_start'
> ```

Hmmm I'm not sure why this happens, need to trace through this...

**Comment by kassane** [comment]
> @kassane could you add/reuse a defconfig to enable hellod ci?

@xiaoxiang781216,
I could try! I will do some experiments on the nuttx IC/CD (I haven't tested on it yet...)

> Hmmm I'm not sure why this happens, need to trace through this... (Oops! Like you mentioned: We need libunwind)

@lupyuen,
Exactly! ldc2 need libunwind for backtracing.

**Pull request merged**